openmc/filter.py

```diff
@@ -1974,10 +1991,22 @@ def from_tabulated1d(cls, tab1d):
         if tab1d.n_regions > 1:
             raise ValueError('Only Tabulated1Ds with a single interpolation '
                              'region are supported')
-        if tab1d.interpolation[0] != 2:
-            raise ValueError('Only linear-linear Tabulated1Ds are supported')
+        if tab1d.interpolation[0] not in (2, 3, 4, 5):
+            raise ValueError('Only linear-linear, linear-log, log-linear, and '
+                             'log-log Tabulated1Ds are supported')
+        out = cls(tab1d.x, tab1d.y)
+
+        # set interpolation type
+        if tab1d.interpolation[0] == 2:
+            out.interpolation = 'linear-linear'
+        elif tab1d.interpolation[0] == 3:
+            out.interpolation = 'linear-log'
+        elif tab1d.interpolation[0] == 4:
+            out.interpolation = 'log-linear'
+        elif tab1d.interpolation[0] == 5:
+            out.interpolation = 'log-log'
 
-        return cls(tab1d.x, tab1d.y)
+        return out
```

Comment:
GitHub won't let me put a suggestion on these lines, but you could use the `INTERPOLATION_SCHEMES` dictionary and then use the argument in the constructor as well, i.e.
```Python
interpolation = cls.INTERPOLATION_SCHEMES[tab1d.interpolation[0]]
return cls(tab1d.x, tab1d.y, interpolation)
```

Reply:
Hah! I just did the same thing before scrolling and seeing this. Thanks!